discovery-provider/src/tasks/entity_manager/social_features.py

```diff
@@ -0,0 +1,175 @@
+            is_current=True,
+            is_delete=False,
+        )
+    if params.action == Action.REPOST:
+        create_record = Repost(
+            blockhash=params.event_blockhash,
+            blocknumber=params.block_number,
+            created_at=params.block_datetime,
+            txhash=params.txhash,
+            user_id=params.user_id,
+            repost_item_id=params.entity_id,
+            repost_type=params.entity_type.lower(),
+            is_current=True,
+            is_delete=False,
+        )
+
+    if create_record:
+        params.add_social_feature_record(
+            params.user_id,
+            params.entity_type,
+            params.entity_id,
+            action_to_record_type[params.action],
+            create_record,
+        )
+
+
+def delete_social_records(params):
+
+    validate_social_feature(params)
+
+    entity_key = (params.user_id, params.entity_type, params.entity_id)
+    record_type = action_to_record_type[params.action]
+    existing_entity = params.existing_records[record_type][entity_key]
+
+    if entity_key in params.new_records[params.entity_type]:
+        # override with last updated playlist is in this block
+        existing_entity = params.new_records[params.entity_type][entity_key][-1]
+
+    deleted_record = None
+    if params.action == Action.UNFOLLOW:
+        deleted_record = Follow(
+            blockhash=params.event_blockhash,
+            blocknumber=params.block_number,
+            follower_user_id=existing_entity.follower_user_id,
+            followee_user_id=existing_entity.followee_user_id,
+            is_current=existing_entity.is_current,
+            is_delete=existing_entity.is_delete,
+            created_at=existing_entity.created_at,
+            txhash=params.txhash,
+        )
+    elif params.action == Action.UNSAVE:
+        deleted_record = Save(
+            blockhash=params.event_blockhash,
+            blocknumber=params.block_number,
+            user_id=existing_entity.user_id,
+            save_item_id=existing_entity.save_item_id,
+            save_type=existing_entity.save_type,
+            is_current=existing_entity.is_current,
+            is_delete=existing_entity.is_delete,
+            created_at=existing_entity.created_at,
+            txhash=params.txhash,
+        )
+    elif params.action == Action.UNREPOST:
+        deleted_record = Repost(
+            blockhash=params.event_blockhash,
+            blocknumber=params.block_number,
+            user_id=existing_entity.user_id,
+            repost_item_id=existing_entity.repost_item_id,
+            repost_type=existing_entity.repost_type,
+            is_current=existing_entity.is_current,
+            is_delete=existing_entity.is_delete,
+            created_at=existing_entity.created_at,
+            txhash=params.txhash,
+        )
+
+    deleted_record.is_delete = True
+
+    if deleted_record:
+        params.add_social_feature_record(
```

Comment:
just calling out that we're calling add_social_feature_record in a delete flow. maybe that's fine though... i guess this is to add a delete record to the db

Reply:
yea i think it's ok since it's still adding a record 🤷